portfolio/src/main/java/com/google/sps/servlets/CuisineDataServlet.java

```diff
@@ -0,0 +1,49 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.sps.data.CuisineDatastore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.google.sps.data.Constants;
+
+@WebServlet("/cuisine-data")
+public final class CuisineDataServlet extends HttpServlet {
+  private static final Logger log = LoggerFactory.getLogger(CuisineDataServlet.class);
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Map<String, Integer> cuisineVotes = CuisineDatastore.fetchCuisineVotes();
+
+    Gson gson = new Gson();
+    String json = gson.toJson(cuisineVotes);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    if (request.getParameter(Constants.CUISINE_PARAMETER) != null) {
+      String cuisine = request.getParameter(Constants.CUISINE_PARAMETER);
```

Comment:
FYI: your list of cuisines is unbounded, so users can enter whatever they want and send it to your API. Your client provides a bounded list (hardcoded) but your server doesn't care. If you are doing any kind of analytics on which cuisines are preferred and such you will probably want more control over what your data is. Note that if you "copy" the javascript cuisines in your server, you need a way to keep them in sync. Typically your server will be the one providing the data to the client so server is your source of truth of available cuisines to vote for.

And if you still want to allow any kind of custom input (users could type whatever) you will want to consider some normalization like lowercasing everything, removing extra spaces (trim), etc. <- in the addCuisineVote